modules/apps/locked-items/locked-items-api/src/main/java/com/liferay/locked/items/renderer/BaseLockedItemsRenderer.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.locked.items.renderer;
+
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * @author Marco Galluzzi
+ */
+public abstract class BaseLockedItemsRenderer implements LockedItemsRenderer {
```

Comment:
Shouldn't this be named `BaseJSPLockedItemsRenderer`? Look for other BaseJSP* classes in portal; this is  a common naming pattern.

Reply:
You are right. More questions about using this pattern, which is not always equally applied (I can find examples of both answers for each question):

# Is it OK to call the main Renderer method `render` instead of `include`?
# Is it OK to make methods, such as `getJspPath`, `protected` instead of `public`?
# Should the subclass `LayoutLockedItemsRenderer` be renamed as `LayoutJSPLockedItemsRenderer`

Reply:
Yes to 1 and 2.

About 3, I don't think so; the fact that LayoutLockedItemsRenderer extends from LayoutJSP... is more of an implementation detail (and that could change in the future). But, if you're super strict about naming conventions, you could argue for that. I'd prefer to not include the JSP chunk in the Layout class.

Reply:
Base class renamed in last commit. Thanks